src/Cake.Core/Tooling/Tool.cs

```diff
@@ -82,7 +82,22 @@ protected void Run(TSettings settings, ProcessArgumentBuilder arguments)
             var process = RunProcess(settings, arguments, processSettings);
 
             // Wait for the process to exit.
-            process.WaitForExit();
+            if (settings.ToolTimeout.HasValue)
+            {
+                if (!process.WaitForExit(settings.ToolTimeout.Value))
+                {
+                    throw new TimeoutException(
+                        string.Format(
```

Comment:
Can we separate formatting as we have done [here](https://github.com/devlead/cake/blob/feature/GH830/src/Cake.Core/Tooling/Tool.cs#L107)?